server/controllers/request.js

```diff
@@ -0,0 +1,103 @@
+const User = require("../models/User");
+const Request = require("../models/Request");
+const asyncHandler = require("express-async-handler");
+const ObjectId = require("mongoose").Types.ObjectId;
+
+// @route GET /requests/id:
+// @desc Get request for logged in user (sitter) by :id
+// @access Private
+exports.loadRequest = asyncHandler(async (req, res, next) => {
+  const id = req.params.id;
+  const request = await Request.findById(id);
+
+  if (request) {
+    res.status(200).send({
+      success: {
+        request,
+      },
+    });
+  } else res.status(400).json({ errors: "Bad request !" });
+});
+
+// @route GET /requests
+// @desc Get requests for logged in user (sitter)
+// @access Private
+exports.listRequests = asyncHandler(async (req, res, next) => {
+  const user = await User.findById(req.user.id);
+  if (!user) {
+    res.status(401);
+    throw new Error("Not authorized");
+  }
+
+  const requests = await Request.find({ userId: ObjectId(user._id) });
+
+  res.status(200).json({
+    success: {
+      requests,
+    },
+  });
+});
+
+// @route POST /requests
+// @desc create new request for logged in user (dog owner)
+// @access Private
+exports.createRequest = asyncHandler(async (req, res, next) => {
+  const { sitterId, dogId, start, end, status } = req.body;
+
+  const request = await Request.create({
+    userId: ObjectId(req.user.id),
+    sitterId: ObjectId(sitterId),
+    dogId: ObjectId(dogId),
+    start,
+    end,
+    status,
+  });
+  if (request) {
+    res.status(201).json({
+      success: {
+        request,
+      },
+    });
+  } else {
+    res.status(400);
```

Comment:
Additionally, since the `create` method calls `save` internally, in the case that any validation fails, the promise returned by `create` should reject and `await` will throw before this block can ever execute. This isn't documented explicitly in the mongoose docs but should be easy to test by sending a request that fails schema validation through an API client such as [Postman](https://www.postman.com/).